internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -234,12 +267,17 @@ func (r *delegatingResolver) updateClientConnStateLocked() error {
 	var endpoints []resolver.Endpoint
 	for _, endpt := range (*r.targetResolverState).Endpoints {
 		var addrs []resolver.Address
-		for _, proxyAddr := range r.proxyAddrs {
-			for _, targetAddr := range endpt.Addresses {
-				addrs = append(addrs, proxyattributes.Set(proxyAddr, proxyattributes.Options{
-					User:        r.proxyURL.User,
-					ConnectAddr: targetAddr.Addr,
-				}))
+		for _, targetAddr := range endpt.Addresses {
+			// Avoid proxy when network is not tcp.
+			if networkType, ok := networktype.Get(targetAddr); ok && networkType != "tcp" {
+				addrs = append(addrs, targetAddr)
+			} else {
+				for _, proxyAddr := range r.proxyAddrs {
+					addrs = append(addrs, proxyattributes.Set(proxyAddr, proxyattributes.Options{
+						User:        r.proxyURL.User,
+						ConnectAddr: targetAddr.Addr,
+					}))
+				}
 			}
```

Comment:
nit: You can use `continue` in the `if` block to avoid indenting the `else` block.